### PR TITLE
gh-127303: Add docs for token.EXACT_TOKEN_TYPES

### PR DESCRIPTION
**Pull request by lubaskinc0de**
* Issue: gh-127303

----
📚 Documentation preview 📚: https://cpython-previews--127304.org.readthedocs.build/

### CODE REVIEW

**Review by Eclips4 — APPROVED** [review]
LGTM. Ilya and I checked that it was added in version 3.8. Although I'm a bit unsure about the wording, I think it could be improved, but neither I nor Ilya are native English speakers :(

File: Doc/library/token.rst [hunk 1]
```diff
@@ -79,6 +79,13 @@ the :mod:`tokenize` module.
    ``type_comments=True``.
 
 
+.. data:: EXACT_TOKEN_TYPES
+
+   A dictionary that stores the string representation of the token
+   as keys, and their numeric codes as values.
```
Comment (tomasr8) [inline Doc/library/token.rst]: How about this?
```suggestion
   A dictionary mapping the string representation of a token to its numeric code.
```

Comment (terryjreedy) [reply Doc/library/token.rst]: done

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Now that it is documented, EXACT_TOKEN_TYPES needs to be added to `__all__` at the top of token.py.  In particular, line 4, which should then be wrapped to 2 lines.
```
__all__ = ['tok_name', 'ISTERMINAL', 'ISNONTERMINAL', 'ISEOF',
           'EXACT_TOKEN_TYPES']
```
EDIT: The complication is that token.py is auto-generated by Tools/build/generate_token.py, so it is the latter that needs revision.  In particular, expand line 229 of that file to 2 lines.

**Comment by lubaskinc0de** [comment]
I have made the requested changes; please review again

File: Doc/library/token.rst [hunk 2]
```diff
@@ -79,6 +79,12 @@ the :mod:`tokenize` module.
    ``type_comments=True``.
 
 
+.. data:: EXACT_TOKEN_TYPES
+
+   A dictionary mapping the string representation of a token to its numeric code.
+
+   .. versionadded:: 3.8
+
```
Comment (picnixz) [inline Doc/library/token.rst:87]: Can we have two blank lines here? (as above)

File: Misc/ACKS [hunk 3]
```diff
@@ -1152,6 +1152,7 @@ Mike Lundy
 Zhongyue Luo
 Mark Lutz
 Taras Lyapun
+Ilya Lyubavski
```
Comment (picnixz) [inline Misc/ACKS]: This should come after Lyn* entries. (Let's try to keep alphabetical order)

File: Misc/NEWS.d/next/Library/2024-11-27-16-06-10.gh-issue-127303.asqkgh.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Expose :data:`token.EXACT_TOKEN_TYPES` in ``__all__``.
```
Comment (picnixz) [inline Misc/NEWS.d/next/Library/2024-11-27-16-06-10.gh-issue-127303.asqkgh.rst]: How about something like this:

```suggestion
Publicly expose :data:`~token.EXACT_TOKEN_TYPES` in :attr:`!token.__all__`.
```

**Review by picnixz — COMMENTED** [review]
While the attribute already existed its scope was not specifically public. Now we can really stress that it's public and documented.

**Review by Eclips4 — APPROVED** [review]
LGTM.

File: Misc/ACKS (hunk 3 above)
Comment (terryjreedy) [reply Misc/ACKS]: Another blank was added.

**Review by terryjreedy — APPROVED** [review]

**Pull request merged**